src/graph.py

```diff
@@ -0,0 +1,110 @@
+"""Graphy mcgraphface"""
+
+class Graph(object):
+    """Define the Link_List class structure"""
+    def __init__(self):
+        """Make an empty dictionary."""
+        self.graph_dict = {}
+
+
+    def add_node(self, value):
+        """Check if node of given value exists in dictionary.
+        If not, add it with an empty list."""
+        try:
+            self.graph_dict[value]
+        except KeyError:
+            self.graph_dict[value] = []
+
+    def add_edge(self, val1, val2):
+        """Ensure that nodes of val1 and val2 exist (creating them if they don't.
+        Then make an edge connecting val1 to val2."""
+        self.add_node(val1)
```

Comment:
What happens if the same edge gets added twice? If I add an edge between 'A' and 'B' you shouldn't have to represent the graph any more than once.